### PR TITLE
test: Add docker kill integration test

### LINKED ISSUE #209: add docker kill integration test
Add docker integration test for kill command

### PR DESCRIPTION
**Pull request by GabyCT**
Add docker integration test for kill command.

Fixes #209

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: integration/docker/kill_test.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+		// http://www.tldp.org/LDP/abs/html/exitcodes.html
+		expectedExitCode += 128
+	}
+
+	return Entry(fmt.Sprintf("with '%d'(%s) signal", signal, syscall.Signal(signal)), signal, expectedExitCode, true)
+}
+
+func withoutSignal() TableEntry {
+	// 137 = 128(command interrupted by a signal) + 9(SIGKILL)
+	return Entry(fmt.Sprintf("without a signal"), syscall.Signal(0), 137, true)
```
Comment (jodh-intel) [inline integration/docker/kill_test.go]: For clarity, I'd do something like this:

```
// Value denoting a command interrupted by a signal (http://www.tldp.org/LDP/abs/html/exitcodes.html)
const interruptedBySignal = 128

  :

expectedExitCode += interruptedBySignal

  : 

expected := interruptedBySignal + int(syscall.SIGKILL)
return Entry(fmt.Sprintf("without a signal"), syscall.Signal(0), expected, true)
```

File: integration/docker/kill_test.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+		Expect(ExistDockerContainer(id)).NotTo(BeTrue())
+	})
+
+	DescribeTable("killing container",
+		func(signal syscall.Signal, expectedExitCode int, waitForExit bool) {
+			args = []string{"--name", id, "-dt", Image, "sh", "-c"}
+
+			switch signal {
+			case syscall.SIGQUIT, syscall.SIGILL, syscall.SIGBUS, syscall.SIGFPE, syscall.SIGSEGV, syscall.SIGPIPE:
+				Skip("This is not forwarded by cc-shim " +
```
Comment (jodh-intel) [inline integration/docker/kill_test.go]: s/cc-shim/kata-shim/

File: integration/docker/kill_test.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+			Expect(exitCode).To(Equal(expectedExitCode))
+		},
+		withSignal(syscall.SIGHUP, canBeTrapped),
+		withSignal(syscall.SIGINT, canBeTrapped),
+		withSignal(syscall.SIGQUIT, canBeTrapped),
+		withSignal(syscall.SIGILL, canBeTrapped),
+		withSignal(syscall.SIGTRAP, canBeTrapped),
+		withSignal(syscall.SIGIOT, canBeTrapped),
+		withSignal(syscall.SIGFPE, canBeTrapped),
+		withSignal(syscall.SIGKILL, cannotBeTrapped), //137
```
Comment (jodh-intel) [inline integration/docker/kill_test.go]: This comment is a bit brief so can you change it somehow? Maybe refer to the `expected` variable idea?

File: integration/docker/kill_test.go (hunk 2 above)
Comment (devimc) [reply integration/docker/kill_test.go]: and move issues 769 and 768 to kata-container/shim

**Comment by GabyCT** [comment]
@jodh-intel changes applied

**Comment by chavafg** [comment]
lgtm

**Pull request merged**